src/pages/workspaces/workspace/Dashboard/CloudInformation.ts

```diff
@@ -0,0 +1,177 @@
+}
+
+interface AzureCloudInformationProps extends CloudInformationProps {
+  workspace: AzureWorkspace & { workspaceInitialized: boolean };
+}
+
+interface GoogleCloudInformationProps extends CloudInformationProps {
+  workspace: GoogleWorkspace & { workspaceInitialized: boolean };
+}
+
+const AzureCloudInformation = (props: AzureCloudInformationProps): ReactNode => {
+  const { workspace, storageDetails } = props;
+  const azureContext = workspace.azureContext;
+  return [
+    dl([h(AzureStorageDetails, { azureContext, storageDetails })]),
+    div({ style: { margin: '0.5rem', fontSize: 12 } }, [
+      div([
+        'Use SAS URL in conjunction with ',
+        h(
+          Link,
+          {
+            ...newTabLinkProps,
+            href: 'https://learn.microsoft.com/en-us/azure/storage/common/storage-use-azcopy-v10',
+            style: { textDecoration: 'underline' },
+          },
+          ['AzCopy']
+        ),
+        ' or ',
+        h(
+          Link,
+          {
+            ...newTabLinkProps,
+            href: 'https://azure.microsoft.com/en-us/products/storage/storage-explorer',
+            style: { textDecoration: 'underline' },
+          },
+          ['Azure Storage Explorer']
+        ),
+        ' to access storage associated with this workspace.',
+      ]),
+      div({ style: { paddingTop: '0.5rem', fontWeight: 'bold' } }, [
+        'The SAS URL expires after 8 hours. To generate a new SAS URL, refresh this page.',
+      ]),
+    ]),
+  ];
+};
+
+const GoogleCloudInformation = (props: GoogleCloudInformationProps): ReactNode => {
+  const { workspace, storageDetails, storageCost, bucketSize } = props;
+  const { accessLevel } = workspace;
+  const { googleProject, bucketName } = workspace.workspace;
+  return [
+    dl([
+      h(InfoRow, { title: 'Cloud Name' }, [
+        h(GcpLogo, { title: 'Google Cloud Platform', role: 'img', style: { height: 16 } }),
+      ]),
+      h(InfoRow, { title: 'Location' }, [h(BucketLocation, { workspace, storageDetails })]),
+      h(InfoRow, { title: 'Google Project ID' }, [
+        h(TooltipCell, [googleProject]),
+        h(ClipboardButton, {
+          'aria-label': 'Copy google project id to clipboard',
+          text: googleProject,
+          style: { marginLeft: '0.25rem' },
+        }),
+      ]),
+      h(InfoRow, { title: 'Bucket Name' }, [
+        h(TooltipCell, [bucketName]),
+        h(ClipboardButton, {
+          'aria-label': 'Copy bucket name to clipboard',
+          text: bucketName,
+          style: { marginLeft: '0.25rem' },
+        }),
+      ]),
+      canWrite(accessLevel) &&
+        h(
+          InfoRow,
+          {
+            title: 'Estimated Storage Cost',
+            subtitle: cond(
+              [!storageCost, () => 'Loading last updated...'],
+              [
+                !!storageCost?.isSuccess,
+                () => `Updated on ${new Date(storageCost?.lastUpdated ?? '').toLocaleDateString()}`,
```

Comment:
If `storageCost?.lastUpdated` is falsy, this will result in "Updated on Invalid Date". It would probably be better to show a different message if `storageCost?.lastUpdated` is falsy.